File: programmers/완전탐색/JS/소수찾기.js
function solution(numbers) {
  return [...new Set(getPer(numbers))].filter((v) => isPrime(v)).length;
}

const getPer = (str) => {
  //str을 기반으로 배열을 만들어라
  const answer = [];
  const n = str.length; //반복문 용도
  const ch = Array.from({ length: n }, () => 0);
  const dfs = (curStr) => {
    //백트래킹 사용
    answer.push(+curStr);
    for (let i = 0; i < n; i++) {
      if (ch[i] === 0) {
        ch[i] = 1;
        dfs(curStr + str[i]);
        ch[i] = 0;
      }
    }
  };

  dfs("");
  answer.shift(); //0은 의미가 없기 때문에 지워버리기
  return answer;
};

//소수 판별하는 함수
const isPrime = (n) => {
  if (n < 2) return false;
  for (let i = 2; i <= Math.sqrt(n); i++) {
    if (n % i === 0) return false;
  }
  return true;
};

File: programmers/완전탐색/Python/소수찾기.py
from itertools import permutations

# 에라스토테네스의 체
def checkPrime(n) :
    if n < 2:
        return False
    for i in range(2, int(n ** 0.5)+1):
        if n % i == 0:
            return False
    return True

def solution(numbers):
    answer = []
    numbers = list(numbers)
    temp = []
    
    for i in range(1, len(numbers) + 1) :
        # numbers 배열에서 i 갯수 만큼 골라 순열을 만들겠다
        temp += list(permutations(numbers, i))
    num = [int(''.join(t)) for t in temp]
        
    for i in num : #num까지 반복문 돌려
        if checkPrime(i) :
            answer.append(i)
            
    return len(set(answer)) #중복 제거하고 최종 배열 길이 출력       